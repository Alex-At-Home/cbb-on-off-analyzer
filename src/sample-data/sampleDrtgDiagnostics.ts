export const sampleDrtgDiagnostics = {
  "stl": 30,
  "blk": 5,
  "drb": 94,
  "pfPct": 0.2,
  "playerRtg": 142.82324406037372,
  "playerDelta": 8.134297736197055,
  "scPossConceded": 0.6341530662717683,
  "noShotCredit": 32.188268933121634,
  "reboundCredit": 36.07039815349106,
  "missFtCredit": 6.070287539936104,
  "stopsIndPct": 0.21044437889736348,
  "stopsTeamPct": 0.15540255483086815,
  "teamBlk": 60,
  "oppoPts": 1804,
  "oppoPoss": 1766,
  "oppoFga": 1429,
  "oppoFgm": 534,
  "oppoFtm": 426,
  "oppoFta": 626,
  "oppoFtPoss": 297.34999999999997,
  "oppoTov": 59,
  "teamStl": 150,
  "opponentOrbPct": 0.2708757637474542,
  "opponentFgPct": 0.3736878936319104,
  "teamOrbCreditToDefender": 0.2724649000411893,
  "teamOrbCreditToRebounder": 0.16965277015673305,
  "teamDvsRebCredit": 0.616272360069244,
  "oppoFgMiss": 835,
  "oppoNonStlTov": -91,
  "teamMissWeight": 0.4376537866243272,
  "oppoFtPct": 0.6805111821086262,
  "oppoFtHitOnePlus": 266.99856230031946,
  "oppoProbFtHitOnePlus": 0.8979268952423726,
  "oppoScPoss": 800.9985623003195,
  "oppoPtsPerScore": 2.2521888114496065,
  "teamRtg": 102.15175537938845,
  "adjDRtg": 102.04282992219363,
  "adjDRtgPlus": 0.4085659844387266,
  "avgEff": 100,
  "dRtg": 110.2860531155855,
  "offSos": 108.0781993204983
 };
