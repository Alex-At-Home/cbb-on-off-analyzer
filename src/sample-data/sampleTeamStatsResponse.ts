export const sampleTeamStatsResponse =
{
  "took" : 73,
  "timed_out" : false,
  "_shards" : {
    "total" : 5,
    "successful" : 5,
    "skipped" : 0,
    "failed" : 0
  },
  "hits" : {
    "total" : {
      "value" : 693,
      "relation" : "eq"
    },
    "max_score" : null,
    "hits" : [ ]
  },
  "aggregations" : {
    "global": {
  		"doc_count": 720,
  		"only": {
  			"buckets": {
  				"team": {
  					"doc_count": 720,
  					"def_3p_opp": {
  						"value": 35.578840579710146
  					},
            "total_off_3p_attempts": {
  						"value": 800
  					},
  					"total_def_3p_attempts": {
  						"value": 709
  					},
            "total_def_2p_attempts": {
  						"value": 900
  					},
  					"total_def_poss": {
  						"value": 2089
  					},
  					"total_def_3p_made": {
  						"value": 241
  					},
  					"def_3p": {
  						"value": 0.33991537376586745
  					},
  					"def_poss": {
  						"value": 2089
  					}
  				}
  			}
  		}
  	},
    "tri_filter" : {
      "buckets" : {
        "baseline" : {
          "doc_count" : 571,
          "total_off_fga" : {
            "value" : 1617.0
          },
          "off_adj_opp" : {
            "value" : 110.32620320855615
          },
          "total_def_2prim_attempts" : {
            "value" : 424.0
          },
          "total_off_2pmid_attempts" : {
            "value" : 485.0
          },
          "total_def_3p_attempts" : {
            "value" : 1000
          },
          "total_off_2_attempts" : {
            "value" : 485 + 424
          },
          "total_off_2pmid_made" : {
            "value" : 206.0
          },
          "total_def_fta" : {
            "value" : 437.0
          },
          "off_2pmidr" : {
            "value" : 0.29993815708101423
          },
          "off_ppp" : {
            "value" : 110.6744556558683
          },
          "off_3pr" : {
            "value" : 0.37105751391465674
          },
          "total_def_fga" : {
            "value" : 1663.0
          },
          "def_poss" : {
            "value" : 1870.0
          },
          "total_off_2prim_made" : {
            "value" : 325.0
          },
          "def_to" : {
            "value" : 0.14385026737967915
          },
          "off_3p" : {
            "value" : 0.3466666666666667
          },
          "def_2p" : {
            "value" : 0.44366197183098594
          },
          "def_adj_opp" : {
            "value" : 98.59978757302179
          },
          "off_2primr" : {
            "value" : 0.329004329004329
          },
          "def_2primr" : {
            "value" : 0.2549609140108238
          },
          "total_off_pts" : {
            "value" : 2084.0
          },
          "off_orb" : {
            "value" : 0.3411513859275053
          },
          "total_def_2pmid_attempts" : {
            "value" : 570.0
          },
          "def_ppp" : {
            "value" : 98.23529411764706
          },
          "def_2pmidr" : {
            "value" : 0.3427540589296452
          },
          "def_3pr" : {
            "value" : 0.40228502705953095
          },
          "total_def_pts" : {
            "value" : 1837.0
          },
          "def_3p" : {
            "value" : 0.3183856502242152
          },
          "def_3p_opp":{"value":36.240722891566264},
          "off_poss" : {
            "value" : 1883.0
          },
          "total_off_2prim_attempts" : {
            "value" : 532.0
          },
          "def_orb" : {
            "value" : 0.24372093023255814
          },
          "total_def_2pmid_made" : {
            "value" : 209.0
          },
          "total_def_2prim_made" : {
            "value" : 232.0
          },
          "off_2p" : {
            "value" : 0.5221238938053098
          },
          "off_efg" : {
            "value" : 0.5213358070500927
          },
          "off_to" : {
            "value" : 0.18693574083908657
          },
          "total_off_fta" : {
            "value" : 530.0
          },
          "def_efg" : {
            "value" : 0.45730607336139506
          },
          "def_2prim" : {
            "value" : 0.5471698113207547
          },
          "off_2pmid" : {
            "value" : 0.4247422680412371
          },
          "def_2pmid" : {
            "value" : 0.36666666666666664
          },
          "off_ftr" : {
            "value" : 0.3277674706246135
          },
          "def_ftr" : {
            "value" : 0.2627781118460613
          },
          "off_2prim" : {
            "value" : 0.6109022556390977
          },
          "off_adj_ppp" : {
            "value" : 120.6744556558683
          },
          "def_adj_ppp" : {
            "value" : 100.23529411764706
          },
          // Assists
          "off_assist" : {
            "value" : 0.50
          },
          "def_assist" : {
            "value" : 0.48
          },
          "total_off_assists": {
            "value": 100
          },
          "total_def_assists": {
            "value": 110
          },
          //(off assist)
          "total_off_ast_3p": {
            "value": 40
          },
          "total_off_ast_mid": {
            "value": 30
          },
          "total_off_ast_rim": {
            "value": 30
          },
          "off_ast_3p": {
            "value": 0.4
          },
          "off_ast_mid": {
            "value": 0.3
          },
          "off_ast_rim": {
            "value": 0.3
          },
          "total_off_3p_ast": {
            "value": 40
          },
          "total_off_2p_ast": {
            "value": 30
          },
          "total_off_2pmid_ast": {
            "value": 30
          },
          "total_off_2prim_ast": {
            "value": 60
          },
          "off_3p_ast": {
            "value": 0.60
          },
          "off_2p_ast": {
            "value": 0.5
          },
          "off_2pmid_ast": {
            "value": 0.25
          },
          "off_2prim_ast": {
            "value": 0.25
          },
          //(def assist)
          "total_def_ast_3p": {
            "value": 45
          },
          "total_def_ast_mid": {
            "value": 35
          },
          "total_def_ast_rim": {
            "value": 35
          },
          "def_ast_3p": {
            "value": 0.45
          },
          "def_ast_mid": {
            "value": 0.35
          },
          "def_ast_rim": {
            "value": 0.35
          },
          "total_def_3p_ast": {
            "value": 45
          },
          "total_def_2p_ast": {
            "value": 35
          },
          "total_def_2pmid_ast": {
            "value": 35
          },
          "total_def_2prim_ast": {
            "value": 65
          },
          "def_3p_ast": {
            "value": 0.65
          },
          "def_2p_ast": {
            "value": 0.55
          },
          "def_2pmid_ast": {
            "value": 0.30
          },
          "def_2prim_ast": {
            "value": 0.30
          },
          //(end assists)
        },
        "off" : {
          "doc_count" : 204,
          "total_off_fga" : {
            "value" : 506.0
          },
          "off_adj_opp" : {
            "value" : 109.2234899328859
          },
          "total_def_2prim_attempts" : {
            "value" : 146.0
          },
          "total_off_2pmid_attempts" : {
            "value" : 157.0
          },
          "total_off_2pmid_made" : {
            "value" : 67.0
          },
          "total_def_fta" : {
            "value" : 135.0
          },
          "off_2pmidr" : {
            "value" : 0.3102766798418972
          },
          "off_ppp" : {
            "value" : 109.15254237288136
          },
          "off_3pr" : {
            "value" : 0.3675889328063241
          },
          "total_def_fga" : {
            "value" : 520.0
          },
          "def_poss" : {
            "value" : 596.0
          },
          "total_off_2prim_made" : {
            "value" : 106.0
          },
          "def_to" : {
            "value" : 0.18120805369127516
          },
          "off_3p" : {
            "value" : 0.3333333333333333
          },
          "def_2p" : {
            "value" : 0.4115853658536585
          },
          "def_adj_opp" : {
            "value" : 99.26440677966102
          },
          "off_2primr" : {
            "value" : 0.3221343873517787
          },
          "def_2primr" : {
            "value" : 0.28076923076923077
          },
          "total_off_pts" : {
            "value" : 644.0
          },
          "off_orb" : {
            "value" : 0.3494809688581315
          },
          "total_def_3p_attempts": {
            "value": 450
          },
          "total_def_2p_attempts": {
            "value": 182 + 163
          },
          "total_def_2pmid_attempts" : {
            "value" : 182.0
          },
          "def_ppp" : {
            "value" : 91.61073825503355
          },
          "def_2pmidr" : {
            "value" : 0.35
          },
          "def_3pr" : {
            "value" : 0.36923076923076925
          },
          "total_def_pts" : {
            "value" : 546.0
          },
          "def_3p" : {
            "value" : 0.3020833333333333
          },
          "def_3p_opp":{"value":37.240722891566264},
          "off_poss" : {
            "value" : 590.0
          },
          "total_off_2prim_attempts" : {
            "value" : 163.0
          },
          "def_orb" : {
            "value" : 0.2577903682719547
          },
          "total_def_2pmid_made" : {
            "value" : 62.0
          },
          "total_def_2prim_made" : {
            "value" : 73.0
          },
          "off_2p" : {
            "value" : 0.540625
          },
          "off_efg" : {
            "value" : 0.525691699604743
          },
          "off_to" : {
            "value" : 0.2016949152542373
          },
          "total_off_fta" : {
            "value" : 154.0
          },
          "def_efg" : {
            "value" : 0.4269230769230769
          },
          "def_2prim" : {
            "value" : 0.5
          },
          "off_2pmid" : {
            "value" : 0.4267515923566879
          },
          "def_2pmid" : {
            "value" : 0.34065934065934067
          },
          "off_ftr" : {
            "value" : 0.30434782608695654
          },
          "def_ftr" : {
            "value" : 0.25961538461538464
          },
          "off_2prim" : {
            "value" : 0.6503067484662577
          },
          "off_adj_ppp" : {
            "value" : 115.6744556558683
          },
          "def_adj_ppp" : {
            "value" : 105.23529411764706
          },
          // Assists
          "off_assist" : {
            "value" : 0.51
          },
          "def_assist" : {
            "value" : 0.49
          },
          "total_off_assists": {
            "value": 101
          },
          "total_def_assists": {
            "value": 111
          },
          //(off assist)
          "total_off_ast_3p": {
            "value": 41
          },
          "total_off_ast_mid": {
            "value": 31
          },
          "total_off_ast_rim": {
            "value": 31
          },
          "off_ast_3p": {
            "value": 0.41
          },
          "off_ast_mid": {
            "value": 0.31
          },
          "off_ast_rim": {
            "value": 0.31
          },
          "total_off_3p_ast": {
            "value": 41
          },
          "total_off_2p_ast": {
            "value": 31
          },
          "total_off_2pmid_ast": {
            "value": 31
          },
          "total_off_2prim_ast": {
            "value": 61
          },
          "off_3p_ast": {
            "value": 0.61
          },
          "off_2p_ast": {
            "value": 0.51
          },
          "off_2pmid_ast": {
            "value": 0.26
          },
          "off_2prim_ast": {
            "value": 0.26
          },
          //(def assist)
          "total_def_ast_3p": {
            "value": 46
          },
          "total_def_ast_mid": {
            "value": 36
          },
          "total_def_ast_rim": {
            "value": 36
          },
          "def_ast_3p": {
            "value": 0.46
          },
          "def_ast_mid": {
            "value": 0.36
          },
          "def_ast_rim": {
            "value": 0.36
          },
          "total_def_3p_ast": {
            "value": 46
          },
          "total_def_2p_ast": {
            "value": 36
          },
          "total_def_2pmid_ast": {
            "value": 36
          },
          "total_def_2prim_ast": {
            "value": 66
          },
          "def_3p_ast": {
            "value": 0.66
          },
          "def_2p_ast": {
            "value": 0.56
          },
          "def_2pmid_ast": {
            "value": 0.31
          },
          "def_2prim_ast": {
            "value": 0.31
          },
          //(end assists)
        },
        "on" : {
          "doc_count" : 367,
          "total_off_fga" : {
            "value" : 1111.0
          },
          "off_adj_opp" : {
            "value" : 110.84207221350077
          },
          "total_def_2prim_attempts" : {
            "value" : 278.0
          },
          "total_off_2pmid_attempts" : {
            "value" : 328.0
          },
          "total_off_2pmid_made" : {
            "value" : 139.0
          },
          "total_def_fta" : {
            "value" : 302.0
          },
          "off_2pmidr" : {
            "value" : 0.2952295229522952
          },
          "off_ppp" : {
            "value" : 111.36890951276102
          },
          "off_3pr" : {
            "value" : 0.3726372637263726
          },
          "total_def_fga" : {
            "value" : 1143.0
          },
          "def_poss" : {
            "value" : 1274.0
          },
          "total_off_2prim_made" : {
            "value" : 219.0
          },
          "def_to" : {
            "value" : 0.12637362637362637
          },
          "off_3p" : {
            "value" : 0.3526570048309179
          },
          "def_2p" : {
            "value" : 0.4594594594594595
          },
          "def_adj_opp" : {
            "value" : 98.29651972157772
          },
          "off_2primr" : {
            "value" : 0.3321332133213321
          },
          "def_2primr" : {
            "value" : 0.2432195975503062
          },
          "total_off_pts" : {
            "value" : 1440.0
          },
          "off_orb" : {
            "value" : 0.337442218798151
          },
          "total_def_3p_attempts": {
            "value": 750
          },
          "total_def_2p_attempts": {
            "value": 650
          },
          "total_def_2pmid_attempts" : {
            "value" : 388.0
          },
          "def_ppp" : {
            "value" : 101.33437990580848
          },
          "def_2pmidr" : {
            "value" : 0.3394575678040245
          },
          "def_3pr" : {
            "value" : 0.41732283464566927
          },
          "total_def_pts" : {
            "value" : 1291.0
          },
          "def_3p" : {
            "value" : 0.3249475890985325
          },
          "def_3p_opp":{"value":35.240722891566264},
          "off_poss" : {
            "value" : 1293.0
          },
          "total_off_3p_attempts": {
            "value": 500
          },
          "total_off_2p_attempts": {
            "value": 600
          },
          "total_off_2prim_attempts" : {
            "value" : 369.0
          },
          "def_orb" : {
            "value" : 0.23684210526315788
          },
          "total_def_2pmid_made" : {
            "value" : 147.0
          },
          "total_def_2prim_made" : {
            "value" : 159.0
          },
          "off_2p" : {
            "value" : 0.5136298421807748
          },
          "off_efg" : {
            "value" : 0.5193519351935193
          },
          "off_to" : {
            "value" : 0.18020108275328692
          },
          "total_off_fta" : {
            "value" : 376.0
          },
          "def_efg" : {
            "value" : 0.47112860892388453
          },
          "def_2prim" : {
            "value" : 0.5719424460431655
          },
          "off_2pmid" : {
            "value" : 0.42378048780487804
          },
          "def_2pmid" : {
            "value" : 0.3788659793814433
          },
          "off_ftr" : {
            "value" : 0.3384338433843384
          },
          "def_ftr" : {
            "value" : 0.2642169728783902
          },
          "off_2prim" : {
            "value" : 0.5934959349593496
          },
          "off_adj_ppp": {
            "value": 111
          },
          "def_adj_ppp": {
            "value": 88
          },
          // Assists
          "off_assist" : {
            "value" : 0.52
          },
          "def_assist" : {
            "value" : 0.50
          },
          "total_off_assists": {
            "value": 102
          },
          "total_def_assists": {
            "value": 112
          },
          //(off assist)
          "total_off_ast_3p": {
            "value": 42
          },
          "total_off_ast_mid": {
            "value": 32
          },
          "total_off_ast_rim": {
            "value": 32
          },
          "off_ast_3p": {
            "value": 0.42
          },
          "off_ast_mid": {
            "value": 0.32
          },
          "off_ast_rim": {
            "value": 0.32
          },
          "total_off_3p_ast": {
            "value": 42
          },
          "total_off_2p_ast": {
            "value": 32
          },
          "total_off_2pmid_ast": {
            "value": 32
          },
          "total_off_2prim_ast": {
            "value": 62
          },
          "off_3p_ast": {
            "value": 0.62
          },
          "off_2p_ast": {
            "value": 0.52
          },
          "off_2pmid_ast": {
            "value": 0.27
          },
          "off_2prim_ast": {
            "value": 0.27
          },
          //(def assist)
          "total_def_ast_3p": {
            "value": 47
          },
          "total_def_ast_mid": {
            "value": 37
          },
          "total_def_ast_rim": {
            "value": 37
          },
          "def_ast_3p": {
            "value": 0.47
          },
          "def_ast_mid": {
            "value": 0.37
          },
          "def_ast_rim": {
            "value": 0.37
          },
          "total_def_3p_ast": {
            "value": 47
          },
          "total_def_2p_ast": {
            "value": 37
          },
          "total_def_2pmid_ast": {
            "value": 37
          },
          "total_def_2prim_ast": {
            "value": 67
          },
          "def_3p_ast": {
            "value": 0.67
          },
          "def_2p_ast": {
            "value": 0.57
          },
          "def_2pmid_ast": {
            "value": 0.32
          },
          "def_2prim_ast": {
            "value": 0.32
          },
          //(end assists)
        }
      }
    }
  }
}
