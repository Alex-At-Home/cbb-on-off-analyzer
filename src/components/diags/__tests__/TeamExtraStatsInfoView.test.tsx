import renderer from 'react-test-renderer';
import React from 'react';
import _ from 'lodash';
import TeamExtraStatsInfoView from '../TeamExtraStatsInfoView';
import { sampleTeamStatsResponse } from "../../../sample-data/sampleTeamStatsResponse"
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

describe("TeamExtraStatsInfoView", () => {
  test("TeamExtraStatsInfoView - should create snapshot", () => {
    const teamData = _.assign(
      sampleTeamStatsResponse.responses[0].aggregations.tri_filter.buckets as { on: any, off: any, baseline: any },
      { global: {}, onOffMode: true }
    );
      const wrapper = shallow(
      <TeamExtraStatsInfoView
        name="test"
        teamStatSet={teamData.baseline}
        showGrades=""
      />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
