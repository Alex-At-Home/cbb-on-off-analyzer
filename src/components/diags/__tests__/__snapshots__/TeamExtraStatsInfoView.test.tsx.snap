// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeamExtraStatsInfoView TeamExtraStatsInfoView - should create snapshot 1`] = `
<span>
  <b>
    Extra stats info for [
    test
    ]
  </b>
  <br />
  <br />
  <a>
    Play type stats:
  </a>
  <Container
    fluid={false}
  >
    <Row
      noGutters={false}
    >
      <Col
        lg={8}
        xs={12}
      >
        <GenericTable
          tableCopyId="playTypeStats_test"
          tableData={
            Array [
              GenericTableDataRow {
                "cellMetaFn": [Function],
                "colSpanOverride": undefined,
                "dataObj": Object {
                  "off_2p": Object {
                    "value": 0.6779661016949152,
                  },
                  "off_3p": Object {
                    "value": 0.30935251798561153,
                  },
                  "off_3pr": Object {
                    "value": 0.439873417721519,
                  },
                  "off_delta_ppp": Object {
                    "value": 0.0905502377016063,
                  },
                  "off_ftr": Object {
                    "value": 0.6708860759493671,
                  },
                  "off_pct": Object {
                    "value": 0.24058738565238325,
                  },
                  "off_pct_orbs": undefined,
                  "off_ppp": Object {
                    "value": 107.2643586151691,
                  },
                  "off_title": "Transition Offense",
                  "off_to": Object {
                    "value": 0.16609965979587754,
                  },
                },
                "kind": "data-row",
                "prefixFn": [Function],
                "tableFieldsOverride": undefined,
              },
              GenericTableDataRow {
                "cellMetaFn": [Function],
                "colSpanOverride": undefined,
                "dataObj": Object {
                  "off_2p": Object {
                    "value": 0.5755395683453237,
                  },
                  "off_3p": Object {
                    "value": 0.28888888888888886,
                  },
                  "off_3pr": Object {
                    "value": 0.24456521739130435,
                  },
                  "off_delta_ppp": Object {
                    "value": 8.046930109156818,
                  },
                  "off_ftr": Object {
                    "value": 0.47282608695652173,
                  },
                  "off_pct": Object {
                    "value": 0.10780838866433026,
                  },
                  "off_pct_orbs": Object {
                    "value": 0.6415989205037649,
                  },
                  "off_ppp": Object {
                    "value": 115.22073848662431,
                  },
                  "off_title": "Scramble Offense",
                  "off_to": Object {
                    "value": 0.12057984260228126,
                  },
                },
                "kind": "data-row",
                "prefixFn": [Function],
                "tableFieldsOverride": Object {
                  "pct": GenericTableColProps {
                    "className": "",
                    "colName": "%",
                    "colorPicker": [Function],
                    "formatter": [Function],
                    "isTitle": false,
                    "missingData": undefined,
                    "rowSpan": [Function],
                    "toolTip": "Percentage of possessions this play type occurs",
                    "widthUnits": 2,
                  },
                },
              },
              GenericTableSeparator {
                "kind": "separator",
              },
              GenericTableDataRow {
                "cellMetaFn": [Function],
                "colSpanOverride": undefined,
                "dataObj": Object {
                  "def_2p": Object {
                    "value": 0.5286624203821656,
                  },
                  "def_3p": Object {
                    "value": 0.35294117647058826,
                  },
                  "def_3pr": Object {
                    "value": 0.4311594202898551,
                  },
                  "def_delta_ppp": Object {
                    "value": -5.395533808941735,
                  },
                  "def_ftr": Object {
                    "value": 0.42028985507246375,
                  },
                  "def_pct": Object {
                    "value": 0.19610764055742433,
                  },
                  "def_pct_orbs": undefined,
                  "def_ppp": Object {
                    "value": 90.66405292820387,
                  },
                  "def_title": "Transition Defense",
                  "def_to": Object {
                    "value": 0.18867924528301885,
                  },
                },
                "kind": "data-row",
                "prefixFn": [Function],
                "tableFieldsOverride": undefined,
              },
              GenericTableDataRow {
                "cellMetaFn": [Function],
                "colSpanOverride": undefined,
                "dataObj": Object {
                  "def_2p": Object {
                    "value": 0.4766355140186916,
                  },
                  "def_3p": Object {
                    "value": 0.42857142857142855,
                  },
                  "def_3pr": Object {
                    "value": 0.24647887323943662,
                  },
                  "def_delta_ppp": Object {
                    "value": 8.196096883864556,
                  },
                  "def_ftr": Object {
                    "value": 0.4507042253521127,
                  },
                  "def_pct": Object {
                    "value": 0.08527071563674632,
                  },
                  "def_pct_orbs": Object {
                    "value": 0.583711708026543,
                  },
                  "def_ppp": Object {
                    "value": 104.25568362101016,
                  },
                  "def_title": "Scramble Defense",
                  "def_to": Object {
                    "value": 0.14088605894731104,
                  },
                },
                "kind": "data-row",
                "prefixFn": [Function],
                "tableFieldsOverride": Object {
                  "pct": GenericTableColProps {
                    "className": "",
                    "colName": "%",
                    "colorPicker": [Function],
                    "formatter": [Function],
                    "isTitle": false,
                    "missingData": undefined,
                    "rowSpan": [Function],
                    "toolTip": "Percentage of possessions this play type occurs",
                    "widthUnits": 2,
                  },
                },
              },
            ]
          }
          tableFields={
            Object {
              "2p": GenericTableColProps {
                "className": "",
                "colName": "2P%",
                "colorPicker": [Function],
                "formatter": [Function],
                "isTitle": false,
                "missingData": undefined,
                "rowSpan": [Function],
                "toolTip": "2 point field goal percentage for this play type",
                "widthUnits": 2,
              },
              "3p": GenericTableColProps {
                "className": "",
                "colName": "3P%",
                "colorPicker": [Function],
                "formatter": [Function],
                "isTitle": false,
                "missingData": undefined,
                "rowSpan": [Function],
                "toolTip": "3 point field goal percentage for this play type",
                "widthUnits": 2,
              },
              "3pr": GenericTableColProps {
                "className": "",
                "colName": "3PR",
                "colorPicker": [Function],
                "formatter": [Function],
                "isTitle": false,
                "missingData": undefined,
                "rowSpan": [Function],
                "toolTip": "Percentage of 3 pointers taken against all field goals for this play type",
                "widthUnits": 2,
              },
              "delta_ppp": GenericTableColProps {
                "className": "",
                "colName": <span>
                  Î”/100
                </span>,
                "colorPicker": [Function],
                "formatter": [Function],
                "isTitle": false,
                "missingData": undefined,
                "rowSpan": [Function],
                "toolTip": "Delta points per 100 possessions between overall play and this play type",
                "widthUnits": 2,
              },
              "ftr": GenericTableColProps {
                "className": "",
                "colName": "FTR",
                "colorPicker": [Function],
                "formatter": [Function],
                "isTitle": false,
                "missingData": undefined,
                "rowSpan": [Function],
                "toolTip": "Free throw rate  for this play type",
                "widthUnits": 2,
              },
              "pct": GenericTableColProps {
                "className": "",
                "colName": "%",
                "colorPicker": [Function],
                "formatter": [Function],
                "isTitle": false,
                "missingData": undefined,
                "rowSpan": [Function],
                "toolTip": "Percentage of possessions ending in this play type",
                "widthUnits": 2,
              },
              "pct_orbs": GenericTableColProps {
                "className": "",
                "colName": "%ORB",
                "colorPicker": [Function],
                "formatter": [Function],
                "isTitle": false,
                "missingData": undefined,
                "rowSpan": [Function],
                "toolTip": "Percentage of Off rebounds resulting in a scramble play type",
                "widthUnits": 2,
              },
              "ppp": GenericTableColProps {
                "className": "",
                "colName": "P/100",
                "colorPicker": [Function],
                "formatter": [Function],
                "isTitle": false,
                "missingData": undefined,
                "rowSpan": [Function],
                "toolTip": "Points per 100 possessions ending in this play type",
                "widthUnits": 2,
              },
              "sep1": GenericTableColProps {
                "className": "",
                "colName": "",
                "colorPicker": [Function],
                "formatter": [Function],
                "isTitle": false,
                "missingData": undefined,
                "rowSpan": [Function],
                "toolTip": "",
                "widthUnits": 0.5,
              },
              "sep2": GenericTableColProps {
                "className": "",
                "colName": "",
                "colorPicker": [Function],
                "formatter": [Function],
                "isTitle": false,
                "missingData": undefined,
                "rowSpan": [Function],
                "toolTip": "",
                "widthUnits": 0.5,
              },
              "sep3": GenericTableColProps {
                "className": "",
                "colName": "",
                "colorPicker": [Function],
                "formatter": [Function],
                "isTitle": false,
                "missingData": undefined,
                "rowSpan": [Function],
                "toolTip": "",
                "widthUnits": 0.5,
              },
              "title": GenericTableColProps {
                "className": "",
                "colName": "",
                "colorPicker": [Function],
                "formatter": [Function],
                "isTitle": true,
                "missingData": undefined,
                "rowSpan": [Function],
                "toolTip": "",
                "widthUnits": 8,
              },
              "to": GenericTableColProps {
                "className": "",
                "colName": "TO%",
                "colorPicker": [Function],
                "formatter": [Function],
                "isTitle": false,
                "missingData": undefined,
                "rowSpan": [Function],
                "toolTip": "Turnover % for this play type",
                "widthUnits": 2,
              },
            }
          }
        />
      </Col>
    </Row>
  </Container>
  <a>
    More assist stats:
  </a>
  <Container
    fluid={false}
  >
    <Row
      noGutters={false}
    >
      <Col
        lg={8}
        xs={12}
      >
        <GenericTable
          tableCopyId="assistStats_test"
          tableData={
            Array [
              GenericTableSubHeaderRow {
                "className": "text-center",
                "cols": Array [
                  "",
                  <i>
                    Assist distribution
                  </i>,
                  "",
                  <i>
                    % of these shots assisted
                  </i>,
                ],
                "kind": "subheader-row",
                "spans": Array [
                  2,
                  3,
                  1,
                  3,
                ],
              },
              GenericTableDataRow {
                "cellMetaFn": [Function],
                "colSpanOverride": undefined,
                "dataObj": Object {
                  "off_3p_ast": Object {
                    "value": 0.4938875305623472,
                  },
                  "off_ast_3p": Object {
                    "value": 0.8487394957983193,
                  },
                  "off_ast_mid": Object {
                    "value": 0.27450980392156865,
                  },
                  "off_ast_rim": Object {
                    "value": 0.4430693069306931,
                  },
                  "off_mid_ast": Object {
                    "value": 0.06845965770171149,
                  },
                  "off_rim_ast": Object {
                    "value": 0.43765281173594134,
                  },
                  "off_title": "Offensive assist details",
                },
                "kind": "data-row",
                "prefixFn": [Function],
                "tableFieldsOverride": undefined,
              },
              GenericTableDataRow {
                "cellMetaFn": [Function],
                "colSpanOverride": undefined,
                "dataObj": Object {
                  "def_3p_ast": Object {
                    "value": 0.485,
                  },
                  "def_ast_3p": Object {
                    "value": 0.8546255506607929,
                  },
                  "def_ast_mid": Object {
                    "value": 0.3254437869822485,
                  },
                  "def_ast_rim": Object {
                    "value": 0.4507462686567164,
                  },
                  "def_mid_ast": Object {
                    "value": 0.1375,
                  },
                  "def_rim_ast": Object {
                    "value": 0.3775,
                  },
                  "def_title": "Defensive assist details",
                },
                "kind": "data-row",
                "prefixFn": [Function],
                "tableFieldsOverride": undefined,
              },
            ]
          }
          tableFields={
            Object {
              "3p_ast": GenericTableColProps {
                "className": "",
                "colName": "3P",
                "colorPicker": [Function],
                "formatter": [Function],
                "isTitle": false,
                "missingData": undefined,
                "rowSpan": [Function],
                "toolTip": "% of assists for 3P",
                "widthUnits": 2,
              },
              "ast_3p": GenericTableColProps {
                "className": "",
                "colName": "3P",
                "colorPicker": [Function],
                "formatter": [Function],
                "isTitle": false,
                "missingData": undefined,
                "rowSpan": [Function],
                "toolTip": "% of assists for 3P",
                "widthUnits": 2,
              },
              "ast_mid": GenericTableColProps {
                "className": "",
                "colName": "Mid",
                "colorPicker": [Function],
                "formatter": [Function],
                "isTitle": false,
                "missingData": undefined,
                "rowSpan": [Function],
                "toolTip": "% of assists for mid-range 2P",
                "widthUnits": 2,
              },
              "ast_rim": GenericTableColProps {
                "className": "",
                "colName": "Mid",
                "colorPicker": [Function],
                "formatter": [Function],
                "isTitle": false,
                "missingData": undefined,
                "rowSpan": [Function],
                "toolTip": "% of assists for 2PAs at the rim",
                "widthUnits": 2,
              },
              "mid_ast": GenericTableColProps {
                "className": "",
                "colName": "Mid",
                "colorPicker": [Function],
                "formatter": [Function],
                "isTitle": false,
                "missingData": undefined,
                "rowSpan": [Function],
                "toolTip": "% of assists for mid-range 2P",
                "widthUnits": 2,
              },
              "rim_ast": GenericTableColProps {
                "className": "",
                "colName": "Mid",
                "colorPicker": [Function],
                "formatter": [Function],
                "isTitle": false,
                "missingData": undefined,
                "rowSpan": [Function],
                "toolTip": "% of assists for 2PAs at the rim",
                "widthUnits": 2,
              },
              "sep1": GenericTableColProps {
                "className": "",
                "colName": "",
                "colorPicker": [Function],
                "formatter": [Function],
                "isTitle": false,
                "missingData": undefined,
                "rowSpan": [Function],
                "toolTip": "",
                "widthUnits": 0.5,
              },
              "sep2": GenericTableColProps {
                "className": "",
                "colName": "",
                "colorPicker": [Function],
                "formatter": [Function],
                "isTitle": false,
                "missingData": undefined,
                "rowSpan": [Function],
                "toolTip": "",
                "widthUnits": 0.5,
              },
              "title": GenericTableColProps {
                "className": "",
                "colName": "",
                "colorPicker": [Function],
                "formatter": [Function],
                "isTitle": true,
                "missingData": undefined,
                "rowSpan": [Function],
                "toolTip": "",
                "widthUnits": 8,
              },
            }
          }
        />
      </Col>
    </Row>
  </Container>
  <a>
    Misc other stats:
  </a>
  <ul>
    <li>
      3P Defensive SoS: [
      <b
        style={
          Object {
            "textShadow": "0px 0px 10px #ffffff,0px 0px 10px #ffffff",
          }
        }
      >
        33.0
      </b>
      ]%
    </li>
    <li>
      Raw tempo: 
      <span>
        <b
          style={
            Object {
              "textShadow": "0px 0px 10px #ffa500,0px 0px 10px #ffa500",
            }
          }
        >
          [
          83.2
          ]
        </b>
         poss/g
      </span>
    </li>
  </ul>
</span>
`;
